cmd/demo/observer-outport.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+
+# Possible values: "server"/"client"
+OBSERVER_MODE="server"
+
+CURRENT_DIR=$(pwd)
+SANDBOX_PATH=$CURRENT_DIR/testnet/testnet-local/sandbox
+KEY_GENERATOR_PATH=$CURRENT_DIR/testnet/mx-chain-go/cmd/keygenerator
+EXTERNAL_CONFIG_DIR=$CURRENT_DIR/exporter/config/external.toml
+
+createObserverKey(){
+  pushd $CURRENT_DIR
+
+  cd testnet/mx-chain-go/cmd/keygenerator
+  go build
+  ./keygenerator
+
+  popd
+}
+
+resetWorkingDir(){
+  rm -rf exporter
+  mkdir "exporter/"
+  cd "exporter/"
+  mkdir "config/"
+}
+
+setupObserver(){
+  OBSERVER_PATH=$(pwd)
+  cp $SANDBOX_PATH/node/node $OBSERVER_PATH
+  cp -R $SANDBOX_PATH/node/config $OBSERVER_PATH
+  mv config/config_observer.toml config/config.toml
+  mv $KEY_GENERATOR_PATH/validatorKey.pem config/
+
+  sed -i "s@DestinationShardAsObserver =.*@DestinationShardAsObserver = \"$1\"@" $OBSERVER_PATH/config/prefs.toml
+  sed -i '/HostDriverConfig\]/!b;n;n;c\    Enabled = true' "$EXTERNAL_CONFIG_DIR"
+  sed -i "s@Mode =.*@Mode = \"$OBSERVER_MODE\"@" "$EXTERNAL_CONFIG_DIR"
+  sed -i 's/MarshallerType =.*/MarshallerType = "json"/' "$EXTERNAL_CONFIG_DIR"
+  sed -i 's/BlockingAckOnError =.*/BlockingAckOnError = false/' "$EXTERNAL_CONFIG_DIR"
+
+  ./node --log-level *:INFO
+}
+
+
+setup(){
+  createObserverKey
+  resetWorkingDir
+  setupObserver $1
```

Comment:
when following the exact same steps from the `demo` readme, I see this warning in the observer:
```
ERROR[2023-05-24 14:53:33.860]   could not start webserver                error = listen tcp 127.0.0.1:8080: bind: address already in use
```
maybe use a different HTTP Rest API port for it

Reply:
Very good catch, thx for trying it out.
I've added extra flag parameter to start it on a different port

Reply:
Instead of reinventing the wheel, why not invoke the script commands from mx-chain-go/scripts/testnet ?
Those will open predefined ports along with having strict & dedicated configs and locations.